src/js/components/Form/__tests__/Form-test.js

```diff
@@ -581,4 +587,80 @@ describe('Form', () => {
     );
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test('uncontrolled without name', () => {
+    const onSubmit = jest.fn();
+    const { getByPlaceholderText, getByText, container } = render(
+      <Form onSubmit={onSubmit}>
+        <FormField>
+          <TextInput placeholder="test input" />
+        </FormField>
+        <Button type="submit" primary label="Submit" />
+      </Form>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'v' },
+    });
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.click(getByText('Submit'));
+    expect(onSubmit).toBeCalled();
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('controlled reset', () => {
+    const onSubmit = jest.fn();
+    const onReset = jest.fn();
+    const Test = () => {
+      const [value, setValue] = React.useState({ test: '' });
+      const onChange = React.useCallback(nextValue => setValue(nextValue), []);
+      return (
+        <Grommet>
+          <Form
+            onReset={onReset}
+            onChange={onChange}
+            value={value}
+            onSubmit={onSubmit}
+          >
+            <FormField name="test" required placeholder="test input" />
+            <Button type="reset" primary label="Reset" />
+          </Form>
+        </Grommet>
+      );
+    };
+    const { getByPlaceholderText, getByText, queryByText } = render(
+      <Grommet>
+        <Test />
+      </Grommet>,
+    );
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'Input has changed' },
+    });
+    fireEvent.click(getByText('Reset'));
+    expect(queryByText('Input has changed')).toBeNull();
+  });
+
+  test('controlled reset without value', () => {
+    const onChange = jest.fn();
+    const Test = () => {
+      return (
+        <Grommet>
+          <Form onChange={onChange}>
+            <FormField name="test" required placeholder="test input" />
+            <Button type="reset" primary label="Reset" />
+          </Form>
+        </Grommet>
+      );
+    };
+    const { getByPlaceholderText, getByText, queryByText } = render(
+      <Grommet>
```

Comment:
Same as above, also it seems a bit redundant to set the component to a variable in the test and then render it, why not just put all of the code in `render` call because it is in the test anyways?

Reply:
Updated so the code is in render instead of test variable. There are certain tests that do need to be set up like this to work but this was not one of them. If the test has hooks (ex. `const [value, setValue] = React.useState({ test: '' });`
 and `const onChange = React.useCallback(nextValue => setValue(nextValue), []);`) then the code has to be outside of render.